### PR TITLE
DOCSP-13849: Modify CRUD op execution

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-13849
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13849-read-write-concern/fundamentals/crud/write-read-pref/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/write-read-pref.txt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+   rc := readconcern.Majority()
+   opts := options.Collection().SetReadConcern(rc)
+   database := client.Database("myDB")
+   coll := database.Collection("myCollection", opts)
+
+Read Preference
+---------------
+
+The read preference option describes how MongoDB clients route read
```
Comment (mongoKart) [inline source/fundamentals/crud/write-read-pref.txt:176]:
```suggestion
The read preference option specifies how the MongoDB client routes read
```

File: source/fundamentals/crud/write-read-pref.txt [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+:rapid:`maxStalenessSeconds </core/read-preference-staleness/>` option, and the
+:rapid:`hedged read </core/read-preference-hedge-option/>` option.
+
+Options
+~~~~~~~
+
+The {+driver-long+} provides the ``readpref`` package to handle
+read preference specification. You can set a read preference using the
+``SetReadPreference()`` method with an ``Option`` type. The ``Option``
+type has the following methods to specify the read concern:
```
Comment (mongoKart) [inline source/fundamentals/crud/write-read-pref.txt:191]: Same comments as above

File: source/fundamentals/crud/write-read-pref.txt [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``Nearest()``
+     - Specifies reads from a random eligible replica set member, irrespective of
+       whether that member is a primary or secondary, based on a specified
```
Comment (mongoKart) [inline source/fundamentals/crud/write-read-pref.txt:202]:
```suggestion
     - Specifies reads from a random eligible replica set member, primary or secondary, based on a specified 
```

File: source/fundamentals/crud/write-read-pref.txt [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``Nearest()``
+     - Specifies reads from a random eligible replica set member, irrespective of
```
Comment (mongoKart) [inline source/fundamentals/crud/write-read-pref.txt:201]: Optional: In all of these descriptions, "Specifies reads" is a little clunky. Could it say "All operations read..." or "Operations read..." or "The client reads..." or something like that?

**Review by mongoKart — CHANGES_REQUESTED** [review]
great job! a few places where things can be clearer or shorter, but if you disagree with anything, please let me know

File: source/fundamentals/crud/write-read-pref.txt [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  unless overridden
+- Session level
+- Transaction level
+- Database level
+- Collection level
+- Query level
+
+You should read this guide if you need to modify how operations execute
+in your replica sets depending on the levels of consistency and
+availability you require from your data.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:36]: tried to shorten the sentence for clarity-- any better?

File: source/fundamentals/crud/write-read-pref.txt [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+replica set member to acknowledge the write operation before the
+operation is deemed successful.
+
+Options
+~~~~~~~
+
+The {+driver-long+} provides the ``writeconcern`` package to handle
+write concern specification. You can set a write concern using the
+``SetWriteConcern()`` method with an ``Option`` type. The ``Option``
+type has the following methods to specify the write concern:
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:54]: see above

File: source/fundamentals/crud/write-read-pref.txt [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+   * - ``WTagSet()``
+     - Requests acknowledgement that write operations propagate to the
+       specified ``mongod`` instance. Takes a ``string`` type. For more
+       information, see the :rapid:`Write Concern specification
+       </reference/write-concern/#mongodb-writeconcern-writeconcern.-custom-write-concern-name->`.
+       
+You can alternatively specify a write concern in your connection string.
+See the :manual:`Server Manual entry on Write Concern Options </reference/connection-string/#write-concern-options>` for more
+information.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:85]: also fixed for read preference

File: source/fundamentals/crud/write-read-pref.txt [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+Example
+~~~~~~~
+
+The following code shows how you can specify a write concern to request
+acknowledgement from two replica set members and create a ``Client``
+with this option:
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:92]: fixed in other sections as well

File: source/fundamentals/crud/write-read-pref.txt [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+   client, err := mongo.Connect(context.TODO(), opts)
+
+Read Concern
+------------
+
+The read concern option allows you to control the consistency and
+isolation properties of the data read from replica sets and replica set
+shards. The default read concern level of "local" returns the instance’s
+most recent data, with no guarantee that the data has been written to a
+majority of the replica set members.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:109]: reworded, but "how you can change it" is still in the options subsection.

File: source/fundamentals/crud/write-read-pref.txt (hunk 4 above)
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt:201]: I changed the Write Concern descriptions as well to match now that all the other sections use this format.

File: source/fundamentals/crud/write-read-pref.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/crud/write-read-pref.txt]: Sorry, I left that TODO for myself as a reminder to come back after I'd read the rest of the page. And I totally forgot. 

The new version sort of seems to suggest that the consistency/availability of the data depends on what the reader does. What about something like:

"You should read this guide if you need to customize the consistency and availability of the data in your replica sets."

?

File: source/fundamentals/crud/write-read-pref.txt [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+operation is deemed successful.
+
+Options
+~~~~~~~
+
+The {+driver-long+} provides the ``writeconcern`` package, which lets
+you specify the write concern for a replica set. Set the write
+concern using the ``SetWriteConcern()`` method with an ``Option``
+type. The ``Option`` type has the following methods to specify the
+read concern:
```
Comment (mongoKart) [inline source/fundamentals/crud/write-read-pref.txt:54]: Write concern?

```suggestion
write concern:
```

File: source/fundamentals/crud/write-read-pref.txt [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+   uri := "mongodb://<hostname>:<port>"
+   wc := writeconcern.W(2)
+   opts := options.Client().ApplyURI(uri).SetWriteConcern(writeconcern.New(wc))
+   client, err := mongo.Connect(context.TODO(), opts)
+
+Read Concern
+------------
+
+The read concern option allows you to control the consistency and
+isolation properties of the data read from replica sets and replica set
```
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/fundamentals/crud/write-read-pref.txt]: Fantastic.

**Review by mongoKart — APPROVED** [review]
Great job with these changes. Two suggestions--one for rewording and one that I think is a typo. I don't need to see it again unless you want to sync on the wording changes

**Pull request merged**